### PR TITLE
bump k8s to 1.21

### PR DESCRIPTION
**Pull request by varshaprasad96**
Co-authored-by: atoato88 <akihito-inou@nec.com>

**What this PR does / why we need it**:
This PR bumps controller runtime to 0.9.0 and k8s dependencies to 1.21.

### CODE REVIEW

**Comment by varshaprasad96** [comment]
Blocked by https://github.com/kubernetes-sigs/controller-runtime/issues/1560

**Comment by atoato88** [comment]
I updated some codes on my branch for passing tests with this PR.
With following changes, I confirmed that `hask/ci/teset.sh` pass all tests in my local env.
https://github.com/atoato88/kubebuilder-declarative-pattern/commit/7ac8645be314300b31939b548067d64db32f5a7c

**Comment by varshaprasad96** [comment]
@atoato88 Thanks a lot! I have included the changes to this PR and added you as a co-author if that's fine. If you would like to open another PR, I can close this one.

**Review by estroz — APPROVED** [review]
/lgtm

**Comment by atoato88** [comment]
> ... If you would like to open another PR, I can close this one.

@varshaprasad96 
Thank you to add me as co-author. It's OK to go forward on this PR :smile: 
"pull-declarative-test" CI job was passed :tada:

**Review by camilamacedo86 — APPROVED** [review]
That is great for me 👍 Good work

**Comment by varshaprasad96** [comment]
@Liujingfang1 @justinsb - pinging here. Would be helpful to have this in!

File: pkg/patterns/declarative/metrics_test.go [hunk 1]
```diff
@@ -529,26 +532,38 @@ func TestAddIfNotPresent(t *testing.T) {
...
+				}
+				_, err = tmpManifests.WriteString(yobj)
+				if err != nil {
+					t.Error(err)
+				}
+				err = tmpManifests.Close()
+				if err != nil {
+					t.Error(err)
+				}
+				defer os.Remove(tmpManifests.Name())
```
Comment (estroz) [inline pkg/patterns/declarative/metrics_test.go:547]: You can use `testing.T.TempDir()` to get a uniquely named test directory that gets cleaned up after the test finishes.
```suggestion
				tmpManifests, err := ioutil.TempFile(t.TempDir(), "tmp-manifests-*.yaml")
				if err != nil {
					t.Error(err)
				}
				_, err = tmpManifests.WriteString(yobj)
				if err != nil {
					tmpManifests.Close()
					t.Error(err)
				}
				err = tmpManifests.Close()
				if err != nil {
					t.Error(err)
				}
```

**Pull request merged**